lib/index.js

```diff
@@ -319,3 +333,134 @@ exports.socket = createSocket;
+      return 'sndbuf set to ' + number + ' bytes'
+    throw new Error(nn.Err() + ': ' + this.type + ' sndbuf@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'sndbuf')
+  }
+}
+
+function rcvbuf(number){
+  if(number){
+    if(setopt(this.binding, 'rcvbuf', number) > -1)
+      return 'rcvbuf set to ' + number + ' bytes'
+    throw new Error(nn.Err() + ': ' + this.type +' rcvbuf@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'rcvbuf')
+  }
+}
+
+function sndtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'sndtimeo', number) > -1)
+      return 'sndtimeo set to ' + number + 'ms'
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'sndtimeo')
+  }
+}
+
+function rcvtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'rcvtimeo', number) > -1)
+      return 'rcvtimeo set to ' + number + 'ms'
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'rcvtimeo')
+  }
+}
+
+function reconn(number){
+  if(number){
+    if(setopt(this.binding, 'reconn', number) > -1)
+      return 'reconn set to ' + number + 'ms'
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'reconn')
+  }
+}
+
+function maxreconn(number){
+  if(number){
+    if(setopt(this.binding, 'maxreconn', number) > -1)
+      return 'maxreconn set to ' + number + 'ms'
+    throw new Error(nn.Err() + ': ' + this.type + ' maxreconn@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'maxreconn')
+  }
+}
+
+function sndprio(number){
+  if(number){
+    if(setopt(this.binding, 'sndprio', number) > -1)
+      return 'sndprio set to ' + number
+    throw new Error(nn.Err() + ': ' + this.type + ' sndprio@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'sndprio')
+  }
+}
+
+function rcvprio(number){
+  if(number){
+    if(setopt(this.binding, 'rcvprio', number) > -1)
+      return 'rcvprio set to ' + number
+    throw new Error(nn.Err() + ': ' + this.type + ' rcvprio@' + number + '\n')
+  } else {
+    return getopt(this.binding, 'rcvprio')
+  }
+}
+
+function tcpnodelay(bool){
+  if(arguments.length){
+    if(bool){
+      if(nn.Setopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY, 1) > -1)
+        return 'tcp nodelay: on'
+      throw new Error(nn.Err() + ': '+this.type + ' nodelay@' + 'activing'+'\n')
+    } else {
+      if(nn.Setopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY, 0) > -1)
+        return 'tcp nodelay: off'
+      throw new Error(nn.Err() + ': '+this.type+' nodelay@' + 'deactiving'+'\n')
+    }
+  } else {
+    switch(nn.Getopt(this.binding, nn.NN_TCP, nn.NN_TCP_NODELAY)){
+      case 1: return 'tcp nodelay: on'
+      case 0: return 'tcp nodelay: off'
+      default:
+        throw new Error(nn.Err() + ': '+this.type+' nodelay@'+'getsockopt'+'\n')
+    }
+  }
+}
```

Comment:
Why does this function return strings of some custom format and not a bool?  Just make it explicit in the docs whether true/false means on/off.